assets/src/stories-editor/components/media-inserter/index.js

```diff
@@ -21,21 +21,25 @@ const POPOVER_PROPS = {
 	position: 'bottom right',
 };
 
-const MediaInserter = ( { insertBlock, updateBlock, canInsertBlockType, showInserter } ) => {
+const MediaInserter = ( { insertBlock, updateBlock, canInsertBlockType, showInserter, getCurrentPageBackgroundType } ) => {
 	const blocks = [
 		'core/video',
 		'core/image',
 	];
+	const mediaType = getCurrentPageBackgroundType();
```

Comment:
Would it not be better to get this information through the `useSelect` hook? As it is now, I don't think this component will re-render if the media type changes.

And if `useSelect` isn't available yet (I'm not sure we're on the right version), at least use the `withSelect` HoC.

Reply:
I think you can remove the prop (and the passing of the prop) and replace this line with:

```suggestion
	const mediaType = useSelect(
		( select ) => {
			const clientId = select( 'amp/story' );
			const blocks = select( 'core/block-editor' ).getBlocksByClientId( clientId );
			return (blocks && blocks.shift().attributes.mediaType) || '';
		}
	);
```

Of course you need to import `useSelect` somewhere.

Reply:
It does use select here https://github.com/ampproject/amp-wp/pull/3109/files#diff-d718b69a2e2ea3f9c4ccfd8999fa59f3R126-R134 .

Reply:
But not `useSelect` which is a hook forcing the component to re-render if the (relevant part of the) store changes.

Reply:
What issue would that solve here?

Reply:
The code as it is has the problem, that if the store updates outside of this component, this component would not update to reflect the correct state – e.g. if the background image is removed via another component, this component would still display "Update Background Image" even though there is no current background image.

Using the `useSelect` hook (or the `withSelect` HoC), React will force this component to re-render with the correct state, regardless of what component or method updates the media type of the current background element.

Reply:
@barklund But this component **does** use `withSelect`.

See the `applyWithSelect` part at the bottom:

https://github.com/ampproject/amp-wp/blob/7451bcc7024fcf0385bdff3d2e13e6bccf6f31aa/assets/src/stories-editor/components/media-inserter/index.js#L112-L137

Reply:
As for `useSelect`, we cannot reliably use it just yet. See https://github.com/ampproject/amp-wp/issues/2538#issuecomment-526165456

Reply:
> @barklund But this component **does** use `withSelect`.

Ah, I did not catch that. Thanks for clarifying. I misunderstood because the argument was given as a callback, not as a direct value.

Is there any reason `mediaType` isn't directly provided as a prop to the component?

Reply:
For consistency reasons I'd also say directly passing `mediaType` makes sense 👍

Reply:
> For consistency reasons I'd also say directly passing `mediaType` makes sense

Do you mean changing from passing a function to passing a variable?

Reply:
Yes

Reply:
Based on your feedback. https://github.com/ampproject/amp-wp/pull/3109/commits/c77650087ac7e82a45f85d85fd8a7a340b266b6f

The feedback about `useSelect` is very valid, as the background can be updated from the sidebar still. Not sure what we can do about this apart from use `useSelect`

Reply:
Since `withSelect` uses `useSelect` under the hood, I don't see an issue here.

Reply:
![Screenshot from 2019-08-30 16-18-19](https://user-images.githubusercontent.com/237508/64032191-d34f1980-cb41-11e9-949f-67c3aeec285f.png)
Screenshot is when I update background from the sidebar. The side should have updated :thinking: